web/js/codeworld_shared.js

```diff
@@ -532,14 +553,49 @@ function registerStandardHints(successFunc) {
       found[2] = [];
     }
 
-    const options = found[0].concat(found[1]).concat(found[2]);
-    for (const candidate of options) {
-      candidate.cost = substitutionCost(
+    const options = found[0].concat(found[1], found[2]);
+
+    options.forEach((candidate) => {
+      const { text } = candidate;
+      const originalTermCost = substitutionCost(
         token.string,
-        candidate.text,
+        text,
         term.length
       );
-    }
+
+      const mappedTerms =
+        Object.prototype.hasOwnProperty.call(replacementTerms.mapping, text) &&
```

Comment:
This looks strangely complex.  Why not just `replacementTerms.mapping.hasOwnProperty(text)`?

Reply:
Yeah, looks odd, but I had to do it since ESLint complained ([link to the respective rule](https://eslint.org/docs/rules/no-prototype-builtins)).

Reply:
I see.  Okay, that looks like there's some justification, so let's leave it that way.